docs/guides/multi-tenant.md

```diff
@@ -0,0 +1,299 @@
+to create a very specifically configured `RDSInstance`, where they can only
+decide the region where it lives and they are restricted to two options.
+
+This model is in contrast to many infrastructure as code tools where the end
+user must have provider credentials to create the underlying resources that are
+rendered from the abstraction. Crossplane takes a different approach, defining
+various credentials in the cluster (using the `ProviderConfig`), then giving
+only the provider controllers the ability to utilize those credentials and
+provision infrastructure on the users behalf. This creates a consistent
+permission model, even when using many providers with differing IAM models, by
+standardizing on Kubernetes RBAC.
+
+### Namespaces as an Isolation Mechanism
+
+While the ability to define abstract schemas and patches to concrete resource
+types using composition is powerful, the ability to define Claim types at the
+namespace scope enhances the functionality further by enabling RBAC to be
+applied with namespace restrictions. Most users in a cluster do not have access
+to cluster-scoped resources as they are considered only relevant to
+infrastructure admins by both Kubernetes and Crossplane.
+
+Building on our simple `CompositeMySQLInstance` / `MySQLInstance` example, a
+platform builder may choose to define permissions on `MySQLInstance` at the
+namespace scope using a `Role`. This allows for giving users the ability to
+create and and manage `MySQLInstances` in their given namespace, but not the
+ability to see those defined in other namespaces.
+
+Futhermore, because the `metadata.namespace` is a field on the XRC, patching can
+be utilized to configure managed resources based on the namespace in which the
+corresponding XRC was defined. This is especially useful if a platform builder
+wants to designate specific credentials or a set of credentials that users in a
+given namespace can utilize when provisioning infrastructure using an XRC. This
+can be accomplished today by creating one or more `ProviderConfig` objects that
+include the name of the namespace in the `ProviderConfig` name. For example, if
+any `MySQLInstance` created in the `team-1` namespace should use specific AWS
+credentials when the provider controller creates the underlying `RDSInstance`,
+the platform builder could:
+
+1. Define a `ProviderConfig` with name `team-1`.
+
+```yaml
+apiVersion: aws.crossplane.io/v1beta1
+kind: ProviderConfig
+metadata:
+  name: team-1
+spec:
+  credentials:
+    source: Secret
+    secretRef:
+      namespace: crossplane-system
+      name: team-1-creds
+      key: key
+```
+
+2. Define a `Composition` that patches the name of the Claim reference in the XR
+   to the `providerConfigRef` of the `RDSInstance`.
+
+```yaml
+...
+resources:
+- base:
+    apiVersion: database.aws.crossplane.io/v1beta1
+    kind: RDSInstance
+    spec:
+      forProvider:
+      ...
+  patches:
+  - fromFieldPath: spec.claimRef.namespace
```

Comment:
It might be worth mentioning the limitation of this approach, i.e. `ProviderConfig` has to be named same as the namespace (or some other, using `map` transform) and you can use only one `ProviderConfig` per namespace.